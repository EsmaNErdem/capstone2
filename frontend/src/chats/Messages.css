.Messages {
    padding: 8px; 
    margin-bottom: 10px; 
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    max-width: 80%;
}

.Messages-sender {
    font-weight: bold;
    margin-right: 8px; 
}