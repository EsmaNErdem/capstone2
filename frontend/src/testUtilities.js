import React from "react";
import UserContext from "./auth/UserContext";

const demoUser = {
  username: "testuser",
  firstNsme: "testfirst",
  lastName: "testlast",
  email: "test@test.net",
  img: null,
};

const UserProvider =
    ({ children, currentUser = demoUser, hasLikedReview = () => false, hasLikedBook = () => false, isUserReview = () => false, hasFollowing = () => false, setCurrentUser = u => u }) => (
    <UserContext.Provider value={{ currentUser, hasLikedReview, hasLikedBook, hasFollowing, setCurrentUser, isUserReview }}>
      {children}
    </UserContext.Provider>
);

const CurrUserProvider =
    ({ children, currentUser = demoUser, hasLikedReview = () => false, hasLikedBook = () => false, isUserReview = () => true, hasFollowing = () => false, setCurrentUser = u => u }) => (
    <UserContext.Provider value={{ currentUser, hasLikedReview, hasLikedBook, hasFollowing, setCurrentUser, isUserReview }}>
      {children}
    </UserContext.Provider>
);

const NonUserProvider =
    ({ children, currentUser = null, hasLikedReview = () => false, hasLikedBook = () => false, hasFollowing = () => false }) => (
    <UserContext.Provider value={{ currentUser, hasLikedReview, hasLikedBook, hasFollowing,}}>
      {children}
    </UserContext.Provider>
);

export { UserProvider, CurrUserProvider, NonUserProvider };
