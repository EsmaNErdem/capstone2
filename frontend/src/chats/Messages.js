import { List, Paper, Typography } from "@mui/material";
import "./Messages.css"

/**
 * Messages Component
 * 
 * - Displays chat message with senders name using Material UI 
 * 
 * - Chat ==> Messages
 */
const Messages = ({ text, username, date=null }) => {
    
    /** Format message timestamp */ 
    const formatMessageDate = (messageDate) => {
        const options = { hour: 'numeric', minute: 'numeric', hour12: true };
        return new Date(messageDate).toLocaleString('en-US', options);
    };

    return (
        <Paper className="Messages" style={{width: '100%', display: 'flex', flexDirection: 'row', justifyContent:'space-between', alignItems: 'center'}}>
            <List>
                <Typography variant="body1" className="Messages-sender">
                    {username}:
                </Typography>
                <Typography variant="body2">{text}</Typography>
            </List>
            {date && <Typography variant="body2">{`${formatMessageDate(date)}`}</Typography>}

        </Paper>
    );
}

export default Messages