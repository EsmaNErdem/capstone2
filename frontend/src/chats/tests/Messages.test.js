import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { UserProvider } from "../../testUtilities";
import { render, fireEvent, act, waitFor } from '@testing-library/react';
import Messages from '../Messages';

  
test('renders without crashing', async () => {  
    await act(async () => {
      render(
        <MemoryRouter>
          <UserProvider>
            <Messages />
          </UserProvider>
        </MemoryRouter>
      );
    });
  
});

test('it renders and matches with snapshot', async () => {  
    await act(async () => {
      ({ asFragment } = render(
        <MemoryRouter>
          <UserProvider>
            <Messages />
          </UserProvider>
        </MemoryRouter>
      ));
    });
  
    expect(asFragment()).toMatchSnapshot();
});

test('it renders and matches with snapshot', async () => {  
    const date = new Date('2024-02-08T12:30:00');

    await act(async () => {
      ({ getByText } = render(
        <MemoryRouter>
          <UserProvider>
            <Messages text={"Hello"} username={"test"} date={date}/>
          </UserProvider>
        </MemoryRouter>
      ));
    });
  
    expect(getByText("12:30 PM")).toBeInTheDocument()
    expect(getByText("test:")).toBeInTheDocument()
    expect(getByText("Hello")).toBeInTheDocument()

});