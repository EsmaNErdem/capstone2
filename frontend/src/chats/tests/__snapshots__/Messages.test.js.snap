// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders and matches with snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 Messages css-1ps6pg7-MuiPaper-root"
    style="width: 100%; display: flex; flex-direction: row; justify-content: space-between; align-items: center;"
  >
    <ul
      class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 Messages-sender css-ahj2mt-MuiTypography-root"
      >
        :
      </p>
      <p
        class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
      />
    </ul>
  </div>
</DocumentFragment>
`;
